### PR TITLE
refactor(core): deprecate hdPath

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Instead of `hdPath(v).derive(p)` use `v.derivePath(p)` or
`v.derivePath(sanitizeLegacyPath(v))` depending on context.

Instead of `deriveKey(p)` use `derivePath(p).keyPair`.

Issue: BG-34381

### CODE REVIEW

File: modules/core/src/travelRule.ts [hunk 1]
```diff
@@ -139,11 +140,9 @@ TravelRule.prototype.decryptReceivedTravelInfo = function (params: DecryptReceiv
     hdNode = bitcoin.HDNode.fromBase58(keychain.xprv);
   }
 
-  const self = this;
-  const hdPathNode = hdPath(hdNode);
-  tx.receivedTravelInfo.forEach(function (info) {
-    const key = hdPathNode.deriveKey(info.toPubKeyPath);
-    const secret = self.bitgo.getECDHSecret({
+  tx.receivedTravelInfo.forEach((info) => {
```
Comment (OttoAllmendinger) [inline modules/core/src/travelRule.ts:143]: we call `this.bitgo.decrypt` further down so I think this is dead code

File: modules/core/src/v2/baseCoin.ts [hunk 2]
```diff
@@ -440,14 +440,17 @@ export abstract class BaseCoin {
    * @returns {{key: string, derivationPath: string}}
    */
   deriveKeyWithSeed({ key, seed }: { key: string; seed: string }): { key: string; derivationPath: string } {
-    const derivationPathInput = bitcoin.crypto.hash256(`${seed}`).toString('hex');
```
Comment (OttoAllmendinger) [inline modules/core/src/v2/baseCoin.ts:443]: `hash256` is double-sha256

File: modules/core/src/bitgo.ts [hunk 3]
```diff
@@ -1045,8 +1046,8 @@ export class BitGo {
 
     // BIP32 derivation path is applied to both client and server master keys
     const derivationPath = responseBody.derivationPath;
-    const clientDerivedNode = hdPath(clientHDNode).derive(derivationPath);
-    const serverDerivedNode = hdPath(serverHDNode).derive(derivationPath);
+    const clientDerivedNode = clientHDNode.derivePath(sanitizeLegacyPath(derivationPath));
+    const serverDerivedNode = serverHDNode.derivePath(sanitizeLegacyPath(derivationPath));
```
Comment (sangaman) [inline modules/core/src/bitgo.ts:1050]: You can call `sanitizeLegacyPath` once and make a `sanitizedDerivationPath` variable right above, saves a call and will make these lines a bit more readable.

**Review by sangaman — APPROVED** [review]

File: modules/core/src/bitgo.ts (hunk 3 above)
Comment (OttoAllmendinger) [reply modules/core/src/bitgo.ts:1050]: true, I will revisit this anyway

**Pull request merged**